core/templates/xregistry_openapi_template.json

```diff
@@ -1,1068 +1,2042 @@
+        "version": "0.5-wip"
+    },
+    "servers": [],
+    "security": [
+        {
+            "api_key": []
         }
-      },
-      "post": {
-        "operationId": "upsertDocument",
-        "description": "Uploads a registry document and upserts its contents into the registry",
-        "requestBody": {
-          "description": "A request to create or update the discovery endpoint's collection of endpoints with the given endpoints",
-          "content": {
-            "application/json": {
-              "schema": {
-                "$ref": "{%-documentTypeReference-%}"
-              }
+    ],
+    "paths": {
+        "/": {
+            "get": {
+                "operationId": "getRootDocument",
+                "description": "Gets the root registry document. Query parameters 'inline', 'filter', 'doc', and 'specversion' modify the view. (See spec: Retrieving the Registry, Inline, Filter, and Doc)",
+                "parameters": [
+                    {
+                        "$ref": "#/components/parameters/inline"
+                    },
+                    {
+                        "$ref": "#/components/parameters/filter"
+                    },
+                    {
+                        "$ref": "#/components/parameters/doc"
+                    },
+                    {
+                        "$ref": "#/components/parameters/specversion"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "The root registry document.",
+                        "content": {
+                            "application/json": {
+                                "schema": {
+                                    "$ref": "#/components/schemas/RegistryEntity"
+                                }
+                            }
+                        }
+                    },
+                    "401": {
+                        "description": "Unauthorized – API key missing or invalid, or authorization failed.",
+                        "content": {
+                            "application/json": {
+                                "schema": {
+                                    "$ref": "#/components/schemas/ProblemDetails"
+                                }
+                            }
+                        }
+                    },
+                    "500": {
+                        "description": "Internal Server Error – an unexpected error occurred on the server.",
+                        "content": {
+                            "application/json": {
+                                "schema": {
+                                    "$ref": "#/components/schemas/ProblemDetails"
+                                }
+                            }
+                        }
+                    }
+                }
+            },
+            "post": {
```

Comment:
POST isn't allowed on the Registry itself. It would need to be on one of the collections.
You can however do a PUT or PATCH at the Registry level. In fact, PATCH is the one I use to import all of your samples w/o overwriting the Registry attributes each time.